multigeiger/multigeiger.ino

```diff
@@ -440,63 +440,30 @@ void loop()
   unsigned int HV_pulse_count;
   char sw[4];
   unsigned long current_ms = millis();                           // to save multiple calls to millis()
-
-  // copy values from ISR
-  portENTER_CRITICAL(&mux_GMC_count);                            // enter critical section
-  GMC_counts = isr_GMC_counts;
-  isr_GMC_counts = 0;
-  count_timestamp = isr_count_timestamp;
-  GMC_counts_2send = isr_GMC_counts_2send;
-  count_timestamp_2send = isr_count_timestamp_2send;
-  portEXIT_CRITICAL(&mux_GMC_count);                             // leave critical section
-
-  // Loop for IoTWebConf
-  iotWebConf.doLoop();
+  bool update_display;
 
   // Read Switches (active LOW!)
   sw[0] = !digitalRead(PIN_SWI_0);
   sw[1] = !digitalRead(PIN_SWI_1);
   sw[2] = !digitalRead(PIN_SWI_2);
   sw[3] = !digitalRead(PIN_SWI_3);
 
-  // make LED flicker and speaker tick
-  if (GMC_counts != last_GMC_counts) {
-    if(ledTick && sw[LED_ON]) {
-      digitalWrite(LED_BUILTIN, HIGH);    // switch on LED
-    }
-
-    if(speakerTick && sw[SPEAKER_ON]) {   // make "Tick" sound
-      for (int speaker_count = 0; speaker_count <= 3; speaker_count++) {
-        digitalWrite (PIN_SPEAKER_OUTPUT_P, LOW);
-        digitalWrite (PIN_SPEAKER_OUTPUT_N, HIGH);
-        delayMicroseconds(500);
-        digitalWrite (PIN_SPEAKER_OUTPUT_P, HIGH);
-        digitalWrite (PIN_SPEAKER_OUTPUT_N, LOW);
-        delayMicroseconds(500);
-      }
-    } else {
-      if(ledTick && sw[LED_ON]) {
-        delay(4);
-      }
-    }
-    if(ledTick && sw[LED_ON]) {
-      digitalWrite(LED_BUILTIN, LOW);     // switch off LED
-    }
-    last_GMC_counts = GMC_counts;         // notice old value
-  }
-
-  // Pulse the high voltage at least every second
-  #define HVPULSERATE 1000
-  if((current_ms - time2hvpulse) >= HVPULSERATE ) {
-    HV_pulse_count = jb_HV_gen_charge__chargepulses();      // charge HV capacitor - restarts time2hvpulse!
-    hvpulsecnt2send += HV_pulse_count;                      // count for sending
-  }
-
   #define DISPLAYREFRESH 10000
   #define MAXCOUNTS 100
+
+  // copy values from ISR
+  portENTER_CRITICAL(&mux_GMC_count);                            // enter critical section
+  GMC_counts = isr_GMC_counts;
+  update_display = (GMC_counts >= MAXCOUNTS) || ((current_ms - time2display) >= DISPLAYREFRESH);
+  if(update_display) isr_GMC_counts = 0;
```

Comment:
this is new.

we only want to reset isr_GMC_counts to 0, IF we update the display.
but we need to do this in a critical section.
thus, compute update_display here (quick) and re-use it in the following if-condition.